docs/dev_guide.md

```diff
@@ -0,0 +1,40 @@
+# Dev Guide
+
+## Adding a new param in GCSFuse
+
+Each param in GCSFuse should be supported as both as a CLI flag as well as via
+config file; the only exception being *--config-file* which is supported only as
+a CLI flag for obvious reasons. Please follow the steps below in order to make a
+new param available in both the modes:
+
+1.  Declare the new param in
+    [params.yaml](https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/cfg/params.yaml#L4).
+    Refer to the documentation at the top of the file for guidance.
+1.  Run `make build` to generate the required code.
```

Comment:
Can we be a bit more explicit, from which directory we need to run the `make build` command?